src/metabase/channel/template/handlebars_helper.clj

```diff
@@ -0,0 +1,123 @@
+(ns metabase.channel.template.handlebars-helper
+  (:refer-clojure :exclude [hash])
+  (:require
+   [java-time.api :as t]
+   [metabase.util.date-2 :as u.date]
+   [metabase.util.urls :as urls])
+  (:import
+   (com.github.jknack.handlebars
+    Options Handlebars Helper Handlebars$SafeString)))
+
+(set! *warn-on-reflection* true)
+
+(defn option-block-body
+  "Get the block body."
+  [^Options option]
+  (.fn option))
+
+(defn option-else-block
+  "Get the else block."
+  [^Options option]
+  (.inverse option))
+
+(defn safe-str
+  "A wrapper of Handlebars.SafeString"
+  [& text]
+  (Handlebars$SafeString. (apply str text)))
+
+(defmacro defhelper
+  "Define a helper function.
+
+  Helper is a function that takes 4 arguments: context, params, hash, and options.
+
+  (defhelper format-name
+    \"Format a name with title and uppercase options.\"
+    [name _params {:keys [title uppercase] :or {title \"Mr.\"}} _options]
+    (if uppercase
+      (str title (u/upper-case-en name))
+      (str title name)))"
+  [helper-name description argvec & body]
+  (let [helper-fn-name (symbol (str helper-name \*))
+        description#   description]
+    (assert (= 4 (count argvec)) "Helper function must have 4 arguments: context, params, hash, and options")
+    `(do
+       (defn ~helper-fn-name
+         ~description#
+         ~argvec
+         ~@body)
+       (def ~(vary-meta helper-name
+                        merge
+                        {:doc        description
+                         :is-helper? true})
+         ~description#
+         (reify Helper (apply [_# context# option#]
+                         (~helper-fn-name context# (.params option#) (update-keys (.hash option#) keyword) option#)))))))
+
+(defn register-helper
+  "Register a helper."
+  [^Handlebars hbs ^String name ^Helper helper]
+  (.registerHelper hbs name helper))
+
+(defn register-helpers
```

Comment:
Should we worry about de-registering old defaults in your hook?

Reply:
hmm, it'd be nice to deregister it, but handlebars does not support deregistering and IMO it's not a big issue. If it poses problem then i think we can replace hbs with a new one in the hook.